department-of-veterans-affairs/next-build
# VACMS-9829 facility menus

## Description
Closes #9829. Adds a recursive menu component for facility menus. Added to the story listing page. Added to storybook.

- based on this existing component: https://github.com/department-of-veterans-affairs/vets-website/tree/0d38b0e6edd941d0ae691d8ad77503adf6c251fd/src/platform/site-wide/side-nav

## Testing done
- pointed my local environment at my local drupal environment and added a menu item underneath "stories" to the VA Altoona Health Care menu.
- added the facility menu to a NewsStory query + component in order to see children render properly

## Screenshots
menu shows only children under the current page, active state
![Screen Shot 2022-08-24 at 3 50 26 PM](https://user-images.githubusercontent.com/11279744/186537406-bfd6909c-22c8-4bf5-8b24-ae832315aa36.png)

example of menu showing on a child page, active state (this is not included in the PR, menu shouldn't appear on individual story pages)
![Screen Shot 2022-08-24 at 3 44 37 PM](https://user-images.githubusercontent.com/11279744/186537399-944f160d-991b-4c34-89c1-4aa152cd0c53.png)



## QA steps
- visit http://localhost:3000/pittsburgh-health-care/stories and see the menu exists.
- visit http://localhost:3000/altoona-health-care/stories and see the menu exists

### Definition of Done
